### PR TITLE
[Improve][Connector-V2] The log outputs detailed exception stack information

### PR DESCRIPTION
**Pull request by lightzhao**
The log outputs detailed exception stack information

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
Now SeaTunnel unified the standard of throwing exception, you can refer to 

https://github.com/apache/incubator-seatunnel/pull/3805/files#diff-6fcfb1fcd6ebc56d8ee2610adeeacfe0b21b7efd86ff12b6a04d30d6d8d2dadbR63 

https://github.com/apache/incubator-seatunnel/issues/2930 get more details.

BTW, before throwing exception does not recommended using `log.error`, this can make log information is a mess.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ private FSDataOutputStream getOrCreateOutputStream(@NonNull String filePath) {
                 beingWrittenOutputStream.put(filePath, fsDataOutputStream);
                 isFirstWrite.put(filePath, true);
             } catch (IOException e) {
-                log.error("can not get output file stream");
+                log.error("can not get output file stream", e);
                 throw new FileConnectorException(CommonErrorCode.FILE_OPERATION_FAILED,
                         String.format("Open file output stream [%s] failed", filePath), e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java]:
```suggestion
                throw new FileConnectorException(CommonErrorCode.FILE_OPERATION_FAILED,
                        String.format("Open file output stream [%s] failed", filePath), e);
```

Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java]: ok, got it. thanks.

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java]: Looking forward your contribution.

**Comment by lightzhao** [comment]
@Hisoka-X @TyrantLucifer please review and merge. thanks.

**Comment by TyrantLucifer** [comment]
> @Hisoka-X @TyrantLucifer please review and merge. thanks.

Deal with this problem:

image

**Comment by lightzhao** [comment]
> > @Hisoka-X @TyrantLucifer please review and merge. thanks.
> 
> Deal with this problem:
> 
> image

done, thanks.

**Comment by lightzhao** [comment]
@TyrantLucifer @Hisoka-X please review.

**Review by TyrantLucifer — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**